src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
     @Override
     public void add(T value) {
-
+        checkAndExpandArray();
+        values[size++] = value;
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        checkBounds(index, size + 1);
+        if (index == size) {
+            add(value);
+        } else {
+            checkAndExpandArray();
+            for (int i = size; i > index; i--) {
+                values[i] = values[i - 1];
+            }
+            values[index] = value;
+            size++;
+        }
```

Comment:
1. In order previous comment to work here we need to swap logic a bit
2. It's better to use structures with as less tabulations of code as possible
3. Use arraycopy method instead of cycle (we can perform copy using same array as source and destination)
```suggestion
    public void add(T value, int index) {
        if (index == size) {
            add(value);
            return;
        }
        checkBounds(index);
        checkAndExpandArray();
        System.arraycopy(....);
        values[index] = value;
        size++;
```